packages/babel-parser/src/tokenizer/index.js

```diff
@@ -917,6 +921,13 @@ export default class Tokenizer extends LocationParser {
             this.raise(this.state.pos, "Invalid or unexpected token");
           }
 
+          if (allowNumSeparator) {
```

Comment:
Shouldn't it only throw if it is _not_ allowed?

Reply:
Oh... My bad. I'll fix it right now.